### PR TITLE
#3 add s block

### LINKED ISSUE #3: Add S block
## User story

```
As a player,
I want to see the "S" block appearing during the game,
So that I can fill an S-like gap with it.
```
## Acceptance criteria

1. The block should be presented on the playfield as:
   ```
    ##
   ##
   
   ```
1. The block is served along with other blocks during the game.

### PR DESCRIPTION
**Pull request by mr-tompkins**
Closes #3

### CODE REVIEW

File: src/test/java/com/epam/prejap/teatrees/block/SBlockTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.epam.prejap.teatrees.block;
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+@Test(groups ="semi-useful")
```
Comment (asilenko) [inline src/test/java/com/epam/prejap/teatrees/block/SBlockTest.java:8]: Why is this group called "semi-useful"?

File: pom.xml [hunk 2]
```diff
@@ -38,7 +38,7 @@
                     <archive>
                         <manifest>
                             <addClasspath>true</addClasspath>
-                            <mainClass>com.epam.prejap.tetris.Tetris</mainClass>
+                            <mainClass>com.epam.prejap.teatrees.TeaTrees</mainClass>
```
Comment (asilenko) [inline pom.xml:41]: https://github.com/EPAM-Online-Courses/teatrees-prejap3/pull/36#pullrequestreview-865848283

File: src/test/java/com/epam/prejap/teatrees/block/SBlockTest.java (hunk 1 above)
Comment (mr-tompkins) [reply src/test/java/com/epam/prejap/teatrees/block/SBlockTest.java:8]: As I understand this tests only checks ones ability to fill an array correctly, but I am open to any suggestions :)

File: pom.xml (hunk 2 above)
Comment (mr-tompkins) [reply pom.xml:41]: Ok, thx I will change that.

File: pom.xml [hunk 3]
```diff
@@ -38,7 +40,7 @@
                     <archive>
                         <manifest>
                             <addClasspath>true</addClasspath>
-                            <mainClass>com.epam.prejap.teatrees.TeaTrees</mainClass>
+                            <mainClass>com.epam.prejap.tetris.Tetris</mainClass>
```
Comment (marcin-ogorzalek-EPAM) [inline pom.xml:43]: Bad change, the original version fixed a bug. You trying to introduce it once more.

File: src/test/java/com/epam/prejap/teatrees/block/SBlockTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import static org.testng.Assert.assertEquals;
+
+@Test(groups = "blocks")
+public class SBlockTest {
+
+    private static final int EXPECTED_HEIGHT = 2;
+    private static final int EXPECTED_WIDTH = 3;
+    private static final byte[][] EXPECTED_DOTS = new byte[][] {{0, 1, 1}, {1, 1, 0}};
+
+    SBlock sBlock;
```
Comment (marcin-ogorzalek-EPAM) [inline src/test/java/com/epam/prejap/teatrees/block/SBlockTest.java:15]: this may be `private`

**Comment by marcin-ogorzalek-EPAM** [comment]
Update project's version in `pom.xml.

**Comment by marcin-ogorzalek-EPAM** [comment]
And sync your branch with the code from `master` in the main repo.

**Review by marcin-ogorzalek-EPAM — CHANGES_REQUESTED** [review]
Close to be done, but there are things to improve

**Review by marcin-ogorzalek-EPAM — APPROVED** [review]
Seems fine. Maybe Tests could be improved but for now, it's ok.

**Review by pkierat-epam — APPROVED** [review]
Looks good.

**Pull request merged**